wordpress-mobile/WordPress-iOS
# Feature/1885 site picker search

Addresses #1885 

Due to the similarities of `BlogListViewController` and `AbstractPostListViewController` I have extracted out the configuration of the `SearchController` and `SearchBar` to a worker class.

The reason I did not add a convenience constructor is because `WPSearchController` is a bridge between `UISearchDisplayController` and `UISearchController`. Any modification to `WPSearchController` that is not in `UISearchController` will have to be re-done. With this solution, the only change required will be to change the parameter type in `init`.

This is my first (largish) commit into iOS, so I'd very much appreciate a nitpicky review so that I'm aware of best practices/conventions/architecture designs I've missed. Thanks in advance!

@aerych @diegoreymendez 

To test:

1. Go to site picker and click the search button.
2. Search bar should appear along w/ the list view refreshing to show all sites (hidden/unhidden)
3. Select a site and it will go into the site details. If the site was previously hidden, it will be unhidden (same behavior as Android)
